src/libpriv/rpmostree-core.c

```diff
@@ -615,16 +614,18 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   if (!dnf_context_setup (self->hifctx, cancellable, error))
     return FALSE;
 
-  /* NB: missing repo --> let hif figure it out for itself */
+  /* NB: missing repo --> let hif figure it out for itself. In that case, it's
+   * also perfectly fine to not have *any* repos at all (eg. local RPM case). */
```

Comment:
Yeah, I was relying on libdnf to throw the error, though it does seem nicer to check and throw this ourselves based on the spec. Done below ⬇️ 

One yucky part is that we're still checking out the tree before throwing this error, but yeah... not today.